test/core/tsi/ssl_transport_security_test.cc

```diff
@@ -681,15 +681,12 @@ int main(int argc, char** argv) {
   ssl_tsi_test_do_handshake_with_server_name_indication_wild_star_domain();
   ssl_tsi_test_do_handshake_with_bad_server_cert();
   ssl_tsi_test_do_handshake_with_bad_client_cert();
-  // TODO: BoringSSL and OpenSSL have different behaviors on handling mismatched
-  // ALPN. Re-enable this test if we can detect in the runtime which SSL library
-  // is used.
-  // ssl_tsi_test_do_handshake_alpn_client_no_server();
+#ifdef OPENSSL_IS_BORINGSSL
```

Comment:
Do you think it would be worth having another test for the other behavior?

Reply:
We do not know whether the mismatched ALPN behavior is on a particular OpenSSL version or all OpenSSL versions. gRPC core does not test OpenSSL extensively. I will create a separate issue to address this, i.e., testing various OpenSSL versions.

Reply:
Sounds good.